### PR TITLE
Maintaining login history of Users

### PR DESCRIPTION
**Pull request by aksshriv1**

### CODE REVIEW

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/common/service/impl/UserLoginHistoryServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+	@Override
+	public UsersLoginHistory createUserLoginHistoryInfo(UserInfo userInfo, String status) {
+		UsersLoginHistory usersLoginHistoryInfo = new UsersLoginHistory();
+		usersLoginHistoryInfo.setUserId(userInfo.getId());
+		usersLoginHistoryInfo.setUserName(userInfo.getUsername());
+		usersLoginHistoryInfo.setEmailId(userInfo.getEmailAddress());
+		usersLoginHistoryInfo.setLoginType(userInfo.getAuthType().toString());
+		usersLoginHistoryInfo.setStatus(status);
+		usersLoginHistoryInfo.setDateAndTime(String.valueOf(Calendar.getInstance().getTime()));
```
Comment (Chittauri) [inline customapi/src/main/java/com/publicissapient/kpidashboard/apis/common/service/impl/UserLoginHistoryServiceImpl.java:30]: use localdatetime

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/common/service/impl/UserLoginHistoryServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
+package com.publicissapient.kpidashboard.apis.common.service.impl;
```
Comment (Chittauri) [inline customapi/src/main/java/com/publicissapient/kpidashboard/apis/common/service/impl/UserLoginHistoryServiceImpl.java:1]: add copyright

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/common/service/impl/UserLoginHistoryServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+import java.util.Calendar;
+
+import org.springframework.beans.factory.annotation.Autowired;
```
Comment (Chittauri) [inline customapi/src/main/java/com/publicissapient/kpidashboard/apis/common/service/impl/UserLoginHistoryServiceImpl.java:5]: create test file for this class

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/common/service/UserLoginHistoryService.java [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package com.publicissapient.kpidashboard.apis.common.service;
```
Comment (Chittauri) [inline customapi/src/main/java/com/publicissapient/kpidashboard/apis/common/service/UserLoginHistoryService.java:1]: add copyright  and comments

File: common/src/main/java/com/publicissapient/kpidashboard/common/repository/rbac/UserLoginHistoryRepository.java [hunk 5]
```diff
@@ -0,0 +1,9 @@
+package com.publicissapient.kpidashboard.common.repository.rbac;
```
Comment (Chittauri) [inline common/src/main/java/com/publicissapient/kpidashboard/common/repository/rbac/UserLoginHistoryRepository.java:1]: add copy right

File: common/src/main/java/com/publicissapient/kpidashboard/common/model/rbac/UsersLoginHistory.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+@AllArgsConstructor
+@NoArgsConstructor
+@Document(collection = "users_login_history")
+public class UsersLoginHistory {
+
+	private ObjectId userId;
+	private String userName;
+	private String emailId;
+	private String loginType;
+	private String dateAndTime;
```
Comment (Chittauri) [inline common/src/main/java/com/publicissapient/kpidashboard/common/model/rbac/UsersLoginHistory.java:24]: change variable name to datetime and confirm the type

File: common/src/main/java/com/publicissapient/kpidashboard/common/model/rbac/UsersLoginHistory.java [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+@Builder
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+@Document(collection = "users_login_history")
+public class UsersLoginHistory {
```
Comment (Chittauri) [inline common/src/main/java/com/publicissapient/kpidashboard/common/model/rbac/UsersLoginHistory.java:18]: extend BasicModel

**Review by hirbabar — APPROVED** [review]

**Review by Chittauri — APPROVED** [review]

**Pull request merged**